### PR TITLE
Remove Delimited workarounds

### PR DESCRIPTION
**Pull request by tunetheweb**
### Brief summary of the change made

Thanks to the excellent work by @WittierDinosaur in #2831 we no longer need to manually construct comma delimited grammars. This PR reverts all the previous use of this pattern.

### Are there any other side effects of this change that we should be aware of?
No. there was one change to YAML file, but this looks to be a good change to me, and previous was incorrect.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: test/fixtures/dialects/tsql/stored_procedure_begin_end.yml [hunk 1]
```diff
@@ -76,8 +76,7 @@ file:
       - parameter: '@out_vchMsg'
       - data_type:
           data_type_identifier: uddt_output_msg
-      - data_type:
-          data_type_identifier: OUTPUT
+      - keyword: OUTPUT
```
Comment (tunetheweb) [inline test/fixtures/dialects/tsql/stored_procedure_begin_end.yml:79]: These are good changes. Should never have been flagged as a data_type before.

**Comment by PhilippLange** [comment]
There is one more instance of this in sparksql (line 849):
```python
Bracketed(
                    # Manually rebuild Delimited.
                    # Delimited breaks complex (MAP, STRUCT) datatypes
                    # (Comma splits angle bracket blocks)
                    Sequence(
                        OneOf(
                            Ref("ColumnDefinitionSegment"),
                            Ref("GeneratedColumnDefinitionSegment"),
                        ),
                        Ref("CommentGrammar", optional=True),
                    ),
```

File: src/sqlfluff/dialects/dialect_teradata.py [hunk 2]
```diff
@@ -450,9 +450,9 @@ class TdCreateTableOptions(BaseSegment):
     """
 
     type = "create_table_options_statement"
-    match_grammar = AnyNumberOf(
-        Sequence(
-            Ref("CommaSegment"),
+    match_grammar = Sequence(
+        Ref("CommaSegment"),
+        Delimited(
             OneOf(
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_teradata.py:456]: Is this OneOf() now unnecessary?

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_teradata.py:456]: Technically yes it is not needed as `Delimited` inherits from `OneOf`.

However, I always forget whether it inherits from `Sequence` or `OneOf` (I had to look it up again) so usually explicitly add the type I want underneath the `Delimited`. A bit of a waste, but more clear IMHO.

WDYT? Think I should just learn this properly and remove this unnecessary `OneOf`?

Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_teradata.py:456]: Technically yes it is not needed as `Delimited` inherits from `OneOf`.

However, I always forget whether it inherits from `Sequence` or `OneOf` (I had to look it up again) so usually explicitly add the type I want underneath the `Delimited`. A bit of a waste, but more clear IMHO.

WDYT? Think I should just learn this properly and remove this unnecessary `OneOf`?

**Comment by tunetheweb** [comment]
> There is one more instance of this in sparksql (line 849):

I thought that one was being taken care of in your #3023 PR and was trying to avoid a merge conflict with that one but see that one was slightly different. Have added it now.

File: src/sqlfluff/dialects/dialect_teradata.py (hunk 2 above)
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_teradata.py:456]: Looks like I'm not the only one! There's 45 instances of `Delimited\(\n\s*OneOf` in our dialects.

As I say, I think it's clearer so would prefer to leave it. But can remove them all if you think it's not a good idea?

**Review by barrywhart — APPROVED** [review]

File: src/sqlfluff/dialects/dialect_redshift.py [hunk 3]
```diff
@@ -1402,31 +1402,18 @@ class ProcedureParameterListSegment(BaseSegment):
...
-                        OneOf(
-                            Ref("ParameterNameSegment"),
-                            exclude=OneOf(_param_type, Ref("ArgModeGrammar")),
-                            optional=True,
-                        ),
-                        Ref("ArgModeGrammar", optional=True),
-                        max_times_per_element=1,
+        Delimited(
+            Sequence(
+                AnyNumberOf(
```
[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/sqlfluff/dialects/dialect_redshift.py:1407]: In this case there is a `_param_type` grammar shortcut as well as the `AnyNumberOf` so it really is a sequence.

**Pull request merged**